web/src/components/FilingsCalendarTaxAccess.test.tsx

```diff
@@ -0,0 +1,338 @@
+import { getMergedConfig } from "@/contexts/configContext";
+import * as api from "@/lib/api-client/apiClient";
+import { generateProfileData, generateUserData, randomLegalStructure } from "@/test/factories";
+import { markdownToText, randomElementFromArray } from "@/test/helpers";
+import {
+  currentUserData,
+  setupStatefulUserDataContext,
+  WithStatefulUserData,
+} from "@/test/mock/withStatefulUserData";
+import { FormationLegalType } from "@businessnjgovnavigator/shared/index";
+import { randomInt } from "@businessnjgovnavigator/shared/intHelpers";
+import { OperatingPhases } from "@businessnjgovnavigator/shared/operatingPhase";
+import { UserData } from "@businessnjgovnavigator/shared/userData";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+import { BusinessPersona } from "../../../shared/src/profileData";
+import { generateFormationData } from "../../test/factories";
+import { FilingsCalendarTaxAccess } from "./FilingsCalendarTaxAccess";
+
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/api-client/apiClient", () => ({
+  postTaxRegistrationOnboarding: jest.fn(),
+}));
+const mockApi = api as jest.Mocked<typeof api>;
+
+const Config = getMergedConfig();
+
+const renderFilingsCalendarTaxAccess = (initialUserData?: UserData) => {
+  render(
+    <WithStatefulUserData initialUserData={initialUserData}>
+      <FilingsCalendarTaxAccess />
+    </WithStatefulUserData>
+  );
+};
+
+describe("<FilingsCalendarTaxAccess />", () => {
+  let userData: UserData;
+  let flow: Exclude<BusinessPersona, undefined>;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    const legalStructure = randomLegalStructure(true);
+    mockApi.postTaxRegistrationOnboarding.mockImplementation(() =>
+      Promise.resolve({ ...userData, taxFilingData: { ...userData.taxFilingData, state: "SUCCESS" } })
+    );
+    userData = generateUserData({
+      profileData: generateProfileData({
+        legalStructureId: legalStructure.id,
+        operatingPhase: randomElementFromArray(
+          OperatingPhases.filter((obj) => obj.displayTaxAccessButton === true)
+        ).id,
+      }),
+      formationData: generateFormationData(
+        { completedFilingPayment: false },
```

Comment:
why are we specifying this? it seems inaccurate, right? By the time a user is accessing tax filings, they have completed and paid for formation. Although, why does this even matter for this test?

Reply:
Part of the AC is that we don't allow users to alter their businessName if they had completed formation with us